docs/DeveloperGuide.md

```diff
@@ -671,29 +673,414 @@ testers are expected to do more *exploratory* testing.
+       Expected: No order is marked as completed. Error details shown in the status message. Status bar remains the same.
+
+### Reverting an order to pending status.
+
+1. Reverting an order to pending status while all orders under a customer are being shown.
+
+    1. Prerequisites:
+        - List all customers using the `listorder 1` command with at least 1 order listed.
+        - The first order's status is not `Pending`.
+
+    2. Test case: `unmarkorder 1`<br>
+       Expected: First order is reverted to pending status. Details of the unmarked order shown in the status message.
+
+    3. Test case: `unmarkorder 0`<br>
+       Expected: No order is reverted to pending status. Error details shown in the status message. Status bar remains the same.
+
+    4. Other incorrect `unmarkorder` commands to try: `unmarkorder`, `unmarkorder x`, `...` (where x is larger than the list size)<br>
+       Expected: Similar to previous.
+
+2. Reverting an order from a filtered list to pending status using the command alias.
+
+    1. Prerequisites: Orders filtered using `filterorder completed` command with at least 1 order listed. <br>
+
+    2. Test case: `unmarko 1`<br>
+       Expected: First order is reverted to pending status. Details of the unmarked order shown in the status message. The unmarked order will disappear from the filtered order list as it is no longer `completed`.
+
+    3. Test case: `unmarko 0`<br>
+       Expected: No order is reverted to pending status. Error details shown in the status message. Status bar remains the same.
+
+3. Attempting to revert a order which currently pending.
+
+    1. Prerequisites:
+        - At least 1 order listed. <br>
+        - The first order's status is `Pending`.
+
+    2. Test case: `unmarko 1`<br>
+       Expected: No order is reverted to pending status. Error details shown in the status message. Status bar remains the same.
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+    1. Prerequisite: You have not edited the `preferences.json` file. There is a folder named `data` in the same directory as the jar file, and there is a `addresbook.json` file in the `data` folder.
+
+    2. Test case: Delete the `addressbook.json` file. Then, run SellSavvy and exit using the `exit` command.<br>
+       Expected: SellSavvy should create a new `addressbook.json` file with default data.
+
+    3. Test case: Delete the `data` folder together with the `addressbook.json` file. Then, run SellSavvy and exit using the `exit` command.<br>
+       Expected: SellSavvy should create a new `data` folder and a new `addressbook.json` file inside the folder with default data.
+
+    4. Test case: Corrupt the `addressbook.json` file by changing its contents to an invalid format, e.g., add a non-numeric character to one of the customer's phone number. Then, run SellSavvy and exit using the `exit` command.<br>
```

Comment:
Not sure if "corrupt" is the right word here, but small nitpick negligible effect.